### PR TITLE
Adds timing metrics for Protocol compute and Event response

### PR DESCRIPTION
**Pull request by jrwils**
https://canvasmedical.atlassian.net/browse/KOALA-1516

### CODE REVIEW

File: plugin_runner/plugin_runner.py [hunk 1]
```diff
@@ -61,12 +68,29 @@ async def HandleEvent(self, request: Event, context):
...
                 log.error(traceback.format_exception(e))
                 continue
             effect_list += effects
 
+        event_duration = get_duration_ms(event_start_time)
+        # Don't log anything if a protocol didn't actually run.
+        if relevant_plugins:
+            log.info(f"Responded to Event {event_name} ({event_duration} ms)")
+            self.statsd_client.timing(
+                f"plugins.event_duration_ms,customer={CUSTOMER_IDENTIFIER},event={event_name}",
```
Comment (beaugunderson) [inline plugin_runner/plugin_runner.py:91]: no need to send `customer`, that is applied by `telegraf.conf` (at the top)

File: plugin_runner/plugin_runner.py [hunk 2]
```diff
@@ -61,12 +68,29 @@ async def HandleEvent(self, request: Event, context):
...
+                compute_start_time = time.time()
                 effects = await asyncio.get_running_loop().run_in_executor(None, protocol.compute)
+                compute_duration = get_duration_ms(compute_start_time)
+                log.info(f"{plugin_name}.compute() completed ({compute_duration} ms)")
+                self.statsd_client.timing(
+                    f"plugins.protocol_duration_ms,customer={CUSTOMER_IDENTIFIER},protocol={plugin_name}",
+                    delta=compute_duration,
+                )
+                # TODO - remove this, just for testing
+                log.info("Sent to statsd")
```
Comment (aduane) [inline plugin_runner/plugin_runner.py:80]: reminder

File: plugin_runner/plugin_runner.py [hunk 3]
```diff
@@ -61,12 +68,29 @@ async def HandleEvent(self, request: Event, context):
...
             try:
                 protocol = protocol_class(request, plugin.get("secrets", {}))
+                compute_start_time = time.time()
                 effects = await asyncio.get_running_loop().run_in_executor(None, protocol.compute)
+                compute_duration = get_duration_ms(compute_start_time)
+                log.info(f"{plugin_name}.compute() completed ({compute_duration} ms)")
+                self.statsd_client.timing(
+                    f"plugins.protocol_duration_ms,plugin={plugin_name}",
+                    delta=compute_duration,
+                )
```
Comment (aduane) [inline plugin_runner/plugin_runner.py:78]: We'll want to replace the `plugin_name` here to include the name of the protocol class. Plugins can include several protocols, we would want to see the timing differences between multiple protocols within a given plugin.

File: plugin_runner/plugin_runner.py (hunk 2 above)
Comment (jrwils) [reply plugin_runner/plugin_runner.py]: Removed.

File: plugin_runner/plugin_runner.py (hunk 3 above)
Comment (jrwils) [reply plugin_runner/plugin_runner.py]: I tested and it's showing both protocols:

```
INFO 2024-06-20 23:41:47,477 In the plugin
INFO 2024-06-20 23:41:50,478 joes_plugin:joes_plugin.protocols.my_protocol:Protocol.compute() completed (3001 ms)
INFO 2024-06-20 23:41:50,479 In the 2nd plugin
INFO 2024-06-20 23:41:51,480 joes_plugin:joes_plugin.protocols.my_2nd_protocol:Protocol.compute() completed (1000 ms)
INFO 2024-06-20 23:41:51,481 Responded to Event ASSESS_COMMAND__CONDITION_SELECTED (4003 ms)
```

**Comment by jrwils** [comment]
@beaugunderson @aduane escaping the colons did the trick. I see data in Grafana now.

**Review by aduane — APPROVED** [review]

**Pull request merged**